impl/src/main/java/org/jboss/weld/event/ObserverNotifier.java

```diff
@@ -432,14 +443,20 @@ private NotificationMode initModeOption(Object value) {
         return () -> {
             final ThreadLocalStackReference<EventMetadata> stack = currentEventMetadata.pushIfNotNull(metadata);
             final RequestContext requestContext = requestContextHolder.get();
+            final FastEvent<Object> requestInitializedEvent = requestInitializedEventHolder.get();
+            final FastEvent<Object> requestBeforeDestroyedEvent = requestBeforeDestroyedEventHolder.get();
+            final FastEvent<Object> requestDestroyedEvent = requestDestroyedEventHolder.get();
             try {
                 securityContext.associate();
+                requestInitializedEvent.fire(toString());
```

Comment:
I think we should use a dedicated `ContextEvent` instance as a payload, similar to  `org.jboss.weld.event.ContextEvent.REQUEST_INITIALIZED_EJB`.